modules/apps/portal-workflow/portal-workflow-test/src/testIntegration/java/com/liferay/portal/workflow/service/test/WorkflowDefinitionLinkServiceImplTest.java

```diff
@@ -0,0 +1,224 @@
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import org.osgi.service.cm.Configuration;
+import org.osgi.service.cm.ConfigurationAdmin;
+
+/**
+ * @author Jhosseph Gonzalez
+ */
+@DataGuard(scope = DataGuard.Scope.METHOD)
+@RunWith(Arquillian.class)
+public class WorkflowDefinitionLinkServiceImplTest {
+
+	@ClassRule
+	@Rule
+	public static final AggregateTestRule aggregateTestRule =
+		new AggregateTestRule(
+			new LiferayIntegrationTestRule(),
+			PermissionCheckerMethodTestRule.INSTANCE);
+
+	@BeforeClass
+	public static void setUpClass() throws Exception {
+		_company = CompanyTestUtil.addCompany();
+
+		PortalInstances.initCompany(_company);
+
+		_companyAdminUser = UserTestUtil.addCompanyAdminUser(_company);
+
+		_configuration = _configurationAdmin.getConfiguration(
+			WorkflowDefinitionConfiguration.class.getName(),
+			StringPool.QUESTION);
+	}
+
+	@Before
+	public void setUp() throws Exception {
+		_originalName = PrincipalThreadLocal.getName();
+		_originalPermissionChecker =
+			PermissionThreadLocal.getPermissionChecker();
+		_serviceContext = ServiceContextTestUtil.getServiceContext();
+	}
+
+	@After
+	public void tearDown() throws Exception {
+		PermissionThreadLocal.setPermissionChecker(_originalPermissionChecker);
+		PrincipalThreadLocal.setName(_originalName);
+
+		ConfigurationTestUtil.saveConfiguration(
+			_configuration,
+			HashMapDictionaryBuilder.<String, Object>put(
+				"company.administrator.can.publish", false
+			).build());
+	}
+
+	@Test
+	public void testAddWorkflowDefinitionLink() throws Exception {
+		KaleoDefinition kaleoDefinition =
+			_kaleoDefinitionLocalService.addKaleoDefinition(
+				null, RandomTestUtil.randomString(),
+				RandomTestUtil.randomString(), RandomTestUtil.randomString(),
+				null, StringPool.BLANK, 1, _serviceContext);
+
+		_kaleoDefinitionLocalService.activateKaleoDefinition(
+			kaleoDefinition.getKaleoDefinitionId(), _serviceContext);
+
+		User user = _addUser();
+
+		_setUpPermissionThreadLocal(user);
+
+		AssertUtils.assertFailure(
+			PrincipalException.MustBeCompanyAdmin.class,
+			StringBundler.concat(
+				"User ", user.getUserId(), " must be the company ",
+				"administrator to perform the action"),
+			() -> _workflowDefinitionLinkService.addWorkflowDefinitionLink(
+				user.getUserId(), TestPropsValues.getCompanyId(),
+				TestPropsValues.getGroupId(), BlogsEntry.class.getName(), 0, 0,
+				kaleoDefinition.getName(), 1));
+
+		_setUpPermissionThreadLocal(_companyAdminUser);
+
+		ConfigurationTestUtil.saveConfiguration(
+			_configuration,
+			HashMapDictionaryBuilder.<String, Object>put(
+				"company.administrator.can.publish", true
+			).build());
+
+		Assert.assertNotNull(
+			_workflowDefinitionLinkService.addWorkflowDefinitionLink(
+				_companyAdminUser.getUserId(), TestPropsValues.getCompanyId(),
+				TestPropsValues.getGroupId(), BlogsEntry.class.getName(), 0, 0,
+				kaleoDefinition.getName(), 1));
+	}
```

Comment:
@JhossephG do [this way](https://github.com/liferay/liferay-portal/blob/71e281da80b10a75a45644c0529dd1de3bacba67/modules/apps/portal-workflow/portal-workflow-kaleo-test/src/testIntegration/java/com/liferay/portal/workflow/kaleo/service/test/KaleoDefinitionServiceImplTest.java#L111) to keep the pattern.

Thank you!